### PR TITLE
fix: fix wrong diagnostics error

### PR DESCRIPTION
**Pull request by amanPras**
fix wrong diagnostics error when an error is introduced in the copybook from two different sources. Closing/opening these file should remove/add appropriate diagnostics.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A

1. Open codea.cbl
2. expect an error on the copybook statement
3. Open diagnos to see the exact error message
4. Now, open codeb.cbl
5. opening codeb.cbl, should not impact the previous diagnostics (now it does)
6. Close codea.cbl, to see the diagnostic disappear

- [x] Test B

1. Open codea.cbl
2. expect an error on the copybook statement
3. Open diagnos to see the exact error message
4. Now, create a new source with the same content as codea.cbl
5. This would result in 2 diagnostics on the copybook with same location and error message
6. Close codea.cbl, to see the diagnostic disappear

codea.cbl
```cobol
0      IDENTIFICATION DIVISION.
1      PROGRAM-ID. TESTREPL.
2      DATA DIVISION.
3      WORKING-STORAGE SECTION.
       copy DIAGNOS REPLACING ==:baaaaa:== by D.
8      PROCEDURE DIVISION.
9          MOVE 0 TO D-TY.

``` 
codeb.cbl
```cobol
0      IDENTIFICATION DIVISION.
1      PROGRAM-ID. TESTREPL.
2      DATA DIVISION.
3      WORKING-STORAGE SECTION.
       copy DIAGNOS REPLACING ==:baaaaa:== by x.
8      PROCEDURE DIVISION.
9          MOVE 0 TO x.

``` 

diagnos.cpy
```cobol
        01 A.
              05 :baaaaa: pic x.    
              05 :baaaaa:-ty pic :baaaaa:x.                                    

``` 

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts [hunk 1]
```diff
@@ -76,6 +77,7 @@ export async function closeActiveEditor() {
     await vscode.commands.executeCommand("undo");
     await sleep(100);
   }
+  await vscode.languages.setTextDocumentLanguage(doc.document, "json");
```
Comment (amanPras) [inline clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts:80]: Closing the editor did not fire the didClose event. This is a hack for the same

File: clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ export async function showDocument(workspace_file: string) {
   const file = await getUri(workspace_file);
   // open and show the file
   const document = await vscode.workspace.openTextDocument(file);
+  await vscode.languages.setTextDocumentLanguage(document, "cobol");
```
Comment (amanPras) [inline clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts:68]: See the comment at line 80

File: clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts (hunk 1 above)
Comment (ishche) [reply clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts:80]: Is this issue only for COBOL files?

Comment (amanPras) [reply clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts:80]: Good point, let me check it for other files as well.

Comment (amanPras) [reply clients/cobol-lsp-vscode-extension/src/test/suite/testHelper.ts:80]: I can see a similar behavior for JSON

**Review by ishche — APPROVED** [review]

**Review by Nurkambay — APPROVED** [review]

**Pull request merged**